src/contentmanager.cpp

```diff
@@ -528,43 +458,24 @@ void ContentManager::openBookPreview(const QString &id)
     } catch (...) {}
 }
 
-namespace
-{
-
-QString downloadStatus2QString(kiwix::Download::StatusResult status)
-{
-    switch(status){
-    case kiwix::Download::K_ACTIVE:   return "active";
-    case kiwix::Download::K_WAITING:  return "waiting";
-    case kiwix::Download::K_PAUSED:   return "paused";
-    case kiwix::Download::K_ERROR:    return "error";
-    case kiwix::Download::K_COMPLETE: return "completed";
-    case kiwix::Download::K_REMOVED:  return "removed";
-    default:                          return "unknown";
-    }
-}
-
-} // unnamed namespace
-
-void ContentManager::downloadStarted(const kiwix::Book& book, const std::string& downloadId)
-{
-    kiwix::Book bookCopy(book);
-    bookCopy.setDownloadId(downloadId);
-    mp_library->addBookBeingDownloaded(bookCopy, getSettingsManager()->getDownloadDir());
-    mp_library->save();
-    emit(oneBookChanged(QString::fromStdString(book.getId())));
-}
-
 void ContentManager::removeDownload(QString bookId)
 {
-    m_downloads.remove(bookId);
-    managerModel->removeDownload(bookId);
+    DownloadManager::removeDownload(bookId);
+    managerModel->setDownloadState(bookId, nullptr);
 }
 
 void ContentManager::downloadDisappeared(QString bookId)
 {
     removeDownload(bookId);
-    kiwix::Book bCopy(mp_library->getBookById(bookId));
+    kiwix::Book bCopy;
+    try {
+        bCopy = mp_library->getBookById(bookId);
```

Comment:
This is the few place where I see getBookById surrounded by try-catch. What is the gurantee that this exception wouldn't happen at the other places in this change that calls getBookById?

Reply:
In the places where `getBookById()` is not protected by `try/catch()` it is called in contexts where the book is supposed to exist. Of course that assumption can be violated in certain non realistic scenarios involving race conditions. But I think that we have many more such "vulnerabilities" in our code.